lib/ruby_lsp/listeners/completion.rb

```diff
@@ -15,15 +15,26 @@ class Completion
           typechecker_enabled: T::Boolean,
           dispatcher: Prism::Dispatcher,
           uri: URI::Generic,
+          trigger_character: T.nilable(String),
         ).void
       end
-      def initialize(response_builder, global_state, node_context, typechecker_enabled, dispatcher, uri) # rubocop:disable Metrics/ParameterLists
+      def initialize( # rubocop:disable Metrics/ParameterLists
+        response_builder,
+        global_state,
+        node_context,
+        typechecker_enabled,
+        dispatcher,
+        uri,
+        trigger_character
+      )
         @response_builder = response_builder
         @global_state = global_state
         @index = T.let(global_state.index, RubyIndexer::Index)
+        @type_inferrer = T.let(global_state.type_inferrer, TypeInferrer)
         @node_context = node_context
         @typechecker_enabled = typechecker_enabled
```

Comment:
Not for this Pr, but I wonder if we can avoid the need to pass `typechecker_enabled` by instead determining this from `type_inferrer`, i.e. if `type_inferrer` is nil that means a typechecker is in use?

Reply:
For now, that would work. However, long term we would like to integrate more deeply with Sorbet and do something like:

1. First ask Sorbet if it can find the most accurate definition
2. If a definition was found, use that and do nothing else
3. If not found, fallback to the Ruby LSP behaviour (we'd need the inferrer here)